### ISSUE TITLE
Model accuracy does not improve

### ISSUE THREAD

**Issue by opfromthestart** [coreylowman/dfdx#715]
This is really a specific question and a general suggestion.
My question is: I have a model at https://github.com/opfromthestart/touhou-ai that uses this library. I am training an autoencoder on a set of images, however the loss does not decrease significantly over time. Even with training on only a single image, it still converges very slowly and does not fully capture the image. I have tried many things, such as adding residuals, getting rid of residuals, adding layers, removing layers, and I don't think any of them seemed to have an effect. My GPU only has 4GB of RAM so I am limited for model size. What could I do to debug why it isn't behaving as well as I would expect?
My suggestion: For any library like this, it is likely that people with little experience will use it for AI without knowing much about it. A wiki page for various training and inference problems not specific to this library, but just to machine learning/neural networks in general may be wanted. This may need to have specific sections for some uses, such as convolution/image and transformer/language processing.

**Comment by coreylowman** [comment]
I see the need for resources on training/debugging, though I think that is a bit out of scope for this library since it is such a huge topic. I've been aiming for anything that is possible in pytorch you can do in dfdx - so ideally all the resources out there for pytorch/tensorflow/other libraries can be re-used for dfdx.

As far as your code, I like the test of training on a single image, because it should memorize that perfectly over time.  I would try the following:

1. Disable all dropout (set all probs to 0 or remove the layers entirely)
2. Verify that the image pixels are actually formatted in the way you expect. For example the popular `image` crate stores images as (W, H, C), which is very different from standard deep learning format of (C, H, W)
3. Verify that gradients are non-zero for all the parameterized layers (you might be seeing vanishing gradient problem)

That's where I'd start at least.

**Comment by coreylowman** [comment]
Closing for now - feel free to reopen with more specific issues
